### DISCUSSION TITLE
Small Site - High Memory Usage ?

### DISCUSSION THREAD

**Discussion by jlmmns** [statamic/cms#7349] [Troubleshooting] [answered]
I run a small site through Forge & DigitalOcean.

I launched the site on a small DO droplet with 1GB of memory.
The site has 9 collections with about 78 entries.

A couple of days after launch, I noticed the memory usage going up pretty fast.
Most of the time it kept fluctuating between 60% and 80%, but also peaking to 100% at times.

First, I increased the PHP memory limit to 1024 MB. (Statamic displays `memory_limit | -1, 1024M`)
And I'm guessing this is why the memory usage started peaking to 100%, haha...

So, I performed a server upgrade from 1GB to 2GB memory.
But the memory usage still kept creeping up, more quickly than I would've thought.

Then I realized the site was still using the local file system for caching.
So I switched to Redis caching, and this had a pretty good impact.

**But the memory usage still keeps creeping up...**

**Is there something else I'm missing, that should get optimized, or excluded from caching?**

Some collection entries (Downloads) hold a handful of bigger files (157 - 195 MB).
**Are those attached files cached as well?**

I've also run `php please stache:refresh` several times, and when deploying through Forge.

The site also runs Horizon (default configuration) for sending contact form mails.
Horizon doesn't seem to be the issue, since it only runs 1 mail job every now and then.

Currently running on Statamic version 3.3.63.

Memory graph from DigitalOcean
<img width="1000" alt="Screen Shot 2023-01-13 at 2 13 11 PM" src="https://user-images.githubusercontent.com/3619890/212416542-1f506b7d-7d0c-4aa6-9a23-0557a8418b67.png">

To summarize, is it normal to have ~55% memory usage, with:
- 2GB Memory
- PHP memory limit of 1024 MB
- Redis caching
- 9 Collections
- 78 Entries

**Comment by duncanmcclean** [comment, marked as answer]
Have you thought about looking into [Static Caching](https://statamic.dev/static-caching#content). I usually use the 'Full' strategy on my sites and it helps a bunch with memory & CPU.

**Reply by jlmmns** [reply]
Thanks for the quick suggestion, Duncan!

Besides the `nocache` tag, is there anything else I need to watch out with, when enabling this?

And considering the 2GB of memory, I'm guessing I should use the "full" measure right away?

**Reply by duncanmcclean** [reply]
If you have anything dynamic on the site (eg. if your dates are relative like `2 days ago`), then you should probably use `{{ nocache }}` around those parts.

You may also need to [configure rules for static caching](https://statamic.dev/static-caching#invalidation) if you need pages other than the one currently being edited to be cleared when entry's are updated

Other than that, I'd enable it and see how you get on. I've only ever used the full measure strategy, haven't had any issues.

**Reply by jlmmns** [reply]
Thanks Duncan! Really appreciate it!

One last thing...
I run a multi-site configuration:
```
'sites' => [

    'nl' => [
        'name' => "Nederlands",
        'locale' => 'nl_BE',
        'url' => env('APP_URL'),
    ],

    'en' => [
        'name' => "English",
        'locale' => 'en_US',
        'url' => env('APP_URL') . '/en',
    ],

],
```

And my contact form uses the `{{ trans }}` tag for translated labels.
But the cached contact page only displays the labels in English.
I've placed the entire form within the `{{ nocache }}` tag.
```
<label class="form-label">{{ trans:labels.name }}</label>
```

When explicitly specifying the locale, the translations remain in the language of the site that got cached first.
```
<label class="form-label">{{ trans:labels.name locale="{site:locale}" }}</label>
```

So there seems to be an issue with static caching and the current locale?

Any idea on how to get this working again?

**Reply by duncanmcclean** [reply]
Sorry, I’m not quite sure about that. I’ve never used multi-site.

**Reply by jlmmns** [reply]
No problem, Duncan!
I'll open this issue in another discussion.

Thanks again!
